crates/ruff/src/rules/pycodestyle/rules/invalid_escape_sequence.rs

```diff
@@ -77,9 +95,35 @@ pub(crate) fn invalid_escape_sequence(
 
         prev = Some(i);
 
-        let Some(next_char) = body[i + 1..].chars().next() else {
+        let next_char = match body[i + 1..].chars().next() {
+            Some(next_char) => next_char,
+            None if tok.is_f_string_middle() => {
+                // If we're at the end of a f-string middle token, the next character
+                // is actually emitted as a different token. For example,
+                //
+                // ```python
+                // f"\{1}"
+                // ```
+                //
+                // is lexed as `FStringMiddle('\\')` and `LBrace`, so we need to check
+                // the next character in the source file.
+                //
+                // Now, if we're at the end of the f-string itself, the lexer wouldn't
+                // have emitted the `FStringMiddle` token in the first place. For example,
+                //
+                // ```python
+                // f"foo\"
+                // ```
+                //
+                // Here, there won't be any `FStringMiddle` because it's an unterminated
+                // f-string.
```

Comment:
I'm a bit confused by this comment. It says that there won't be any `FStringMiddle` token but this branch handles f-string middle tokens. How is this related?

Reply:
That is for the code snippet mentioned above. I'm making a case to prove that the next character is always going to be inside the f-string as a whole. The code snippet is:

```python
f"foo\"
```

Here, the `FStringMiddle` token won't be emitted because it's an unterminated string literal as the end quote is being escaped. So, the character coming after a `\` is always going to be part of the f-string.